<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Ryan Jensen">

<title>Mathnotes - Protein Absorption in Cells</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<link href="../../projects/datasaurus/index.html" rel="next">
<link href="../../projects/index.html" rel="prev">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-03CF79D67N"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-03CF79D67N', { 'anonymize_ip': true});
</script>
<link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/gh/aaaakshat/cm-web-fonts@latest/fonts.css">


<link rel="stylesheet" href="../../styles.css">
</head>

<body class="nav-sidebar floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Mathnotes</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="../../index.html" rel="" target="" aria-current="page">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../python/index.html" rel="" target="">
 <span class="menu-text">Python</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../julia/index.html" rel="" target="">
 <span class="menu-text">Julia</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../ashlyn-math/index.html" rel="" target="">
 <span class="menu-text">Ashlyn Math</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../projects/index.html" rel="" target="">
 <span class="menu-text">Research Projects</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../resume.html" rel="" target=""><i class="bi bi-person-vcard" role="img">
</i> 
 <span class="menu-text">Resume</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/ryan-jensen/mathnotes" rel="" target=""><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../projects/index.html">Research Projects</a></li><li class="breadcrumb-item"><a href="../../projects/cells/index.html">Protein Absorption in Cells</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-center sidebar-header">
      <a href="../../index.html" class="sidebar-logo-link">
      <img src="../../mathnotes-logo.svg" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../../projects/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Research Projects</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../projects/cells/index.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Protein Absorption in Cells</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../projects/datasaurus/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Datasaurus</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../projects/level-sets/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Homology of Level Sets</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of Contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="header-section-number">1</span> Introduction</a></li>
  <li><a href="#imports" id="toc-imports" class="nav-link" data-scroll-target="#imports"><span class="header-section-number">2</span> Imports</a></li>
  <li><a href="#helper-function" id="toc-helper-function" class="nav-link" data-scroll-target="#helper-function"><span class="header-section-number">3</span> Helper Function</a></li>
  <li><a href="#image-processing-function" id="toc-image-processing-function" class="nav-link" data-scroll-target="#image-processing-function"><span class="header-section-number">4</span> Image Processing Function</a></li>
  <li><a href="#test-image" id="toc-test-image" class="nav-link" data-scroll-target="#test-image"><span class="header-section-number">5</span> Test Image</a>
  <ul class="collapse">
  <li><a href="#read-and-process-the-image" id="toc-read-and-process-the-image" class="nav-link" data-scroll-target="#read-and-process-the-image"><span class="header-section-number">5.1</span> Read and Process the Image</a></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results"><span class="header-section-number">5.2</span> Results</a></li>
  </ul></li>
  <li><a href="#high-intensity-on-the-cell-boundary" id="toc-high-intensity-on-the-cell-boundary" class="nav-link" data-scroll-target="#high-intensity-on-the-cell-boundary"><span class="header-section-number">6</span> High Intensity on the Cell Boundary</a>
  <ul class="collapse">
  <li><a href="#read-and-process-the-image-1" id="toc-read-and-process-the-image-1" class="nav-link" data-scroll-target="#read-and-process-the-image-1"><span class="header-section-number">6.1</span> Read and Process the Image</a></li>
  <li><a href="#raw-image-slices-1" id="toc-raw-image-slices-1" class="nav-link" data-scroll-target="#raw-image-slices-1"><span class="header-section-number">6.2</span> Raw Image Slices</a></li>
  <li><a href="#d-point-cloud-1" id="toc-d-point-cloud-1" class="nav-link" data-scroll-target="#d-point-cloud-1"><span class="header-section-number">6.3</span> 3D Point Cloud</a></li>
  <li><a href="#the-persistence-diagram-1" id="toc-the-persistence-diagram-1" class="nav-link" data-scroll-target="#the-persistence-diagram-1"><span class="header-section-number">6.4</span> The Persistence Diagram</a></li>
  </ul></li>
  <li><a href="#medium-intensity-on-the-cell-boundary" id="toc-medium-intensity-on-the-cell-boundary" class="nav-link" data-scroll-target="#medium-intensity-on-the-cell-boundary"><span class="header-section-number">7</span> Medium Intensity on the Cell Boundary</a></li>
  <li><a href="#low-intensity-on-the-cell-boundary" id="toc-low-intensity-on-the-cell-boundary" class="nav-link" data-scroll-target="#low-intensity-on-the-cell-boundary"><span class="header-section-number">8</span> Low Intensity on the Cell Boundary</a></li>
  <li><a href="#results-comparison" id="toc-results-comparison" class="nav-link" data-scroll-target="#results-comparison"><span class="header-section-number">9</span> Results Comparison</a></li>
  </ul>
<div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="index.pdf"><i class="bi bi-file-pdf"></i>PDF</a></li><li><a href="index.md"><i class="bi bi-file-code"></i>CommonMark</a></li><li><a href="index.org"><i class="bi bi-file"></i>Org-Mode</a></li><li><a href="index.ipynb" download="index.ipynb"><i class="bi bi-journal-code"></i>Jupyter</a></li><li><a href="index.adoc"><i class="bi bi-file"></i>Asciidoc</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Protein Absorption in Cells</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Ryan Jensen </p>
          </div>
  </div>
    
  
    
  </div>
  

</header>

<section id="introduction" class="level2" data-number="1">
<h2 data-number="1" class="anchored" data-anchor-id="introduction"><span class="header-section-number">1</span> Introduction</h2>
<p>What follows is a proof of concept for using persistent homology to establish the effectiveness of medication in treating sick cells. Cells are sick in the sense that they cannot absorb a certain protein. The medication is suppose to allow the protein to be absorbed. The protein has been dyed so as to show up in images. The cells are grouped and each group is given a medication, next the protein is introduced to the cell. Effective medication have the protein distributed throughout the cell. Ineffective medication have the protein concentrated along the cell boundary. The problem is to find a way to compute the effectiveness of the various medications.</p>
</section>
<section id="imports" class="level2" data-number="2">
<h2 data-number="2" class="anchored" data-anchor-id="imports"><span class="header-section-number">2</span> Imports</h2>
<p>Here we import the needed libraries. We are suppressing the messages from loading the libraries to keep the resulting html document cleaner. We first check if the package `install.load` is installed, if not we install and load it, if it is installed we simply load it. Next we use this library to install (if needed) and load the renaming libraries.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode numberSource R code number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">suppressMessages</span>({</span>
<span id="cb1-2"><a href="#cb1-2"></a>  <span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(<span class="st">"install.load"</span>)) {</span>
<span id="cb1-3"><a href="#cb1-3"></a>  <span class="fu">install.packages</span>(<span class="st">"install.load"</span>)</span>
<span id="cb1-4"><a href="#cb1-4"></a>  }</span>
<span id="cb1-5"><a href="#cb1-5"></a>})</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="fu">suppressMessages</span>(<span class="fu">install_load</span>(<span class="st">"tiff"</span>))      <span class="co"># Using: readTIFF</span></span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="fu">suppressMessages</span>(<span class="fu">install_load</span>(<span class="st">"rtiff"</span>))     <span class="co"># Using: autoThreshold</span></span>
<span id="cb1-8"><a href="#cb1-8"></a><span class="fu">suppressMessages</span>(<span class="fu">install_load</span>(<span class="st">"RSAGA"</span>))     <span class="co"># Using: grid.to.xyz</span></span>
<span id="cb1-9"><a href="#cb1-9"></a><span class="fu">suppressMessages</span>(<span class="fu">install_load</span>(<span class="st">"TDA"</span>))       <span class="co"># Using: ripsDiag, </span></span>
<span id="cb1-10"><a href="#cb1-10"></a><span class="fu">suppressMessages</span>(<span class="fu">install_load</span>(<span class="st">"plotly"</span>))    <span class="co"># Using: plot_ly</span></span>
<span id="cb1-11"><a href="#cb1-11"></a><span class="fu">suppressMessages</span>(<span class="fu">install_load</span>(<span class="st">"plot3D"</span>))    <span class="co"># Using: image2D</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="helper-function" class="level2" data-number="3">
<h2 data-number="3" class="anchored" data-anchor-id="helper-function"><span class="header-section-number">3</span> Helper Function</h2>
<p>This function applies auto-thresholding to its input, which is an image. If a pixel value is less than the threshold, it replaced by 0. The algorithm for determining the threshold value is in the package `rtiff`.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode numberSource R code number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>autoTH <span class="ot">&lt;-</span> <span class="cf">function</span>(x) </span>
<span id="cb2-2"><a href="#cb2-2"></a>{</span>
<span id="cb2-3"><a href="#cb2-3"></a>  th <span class="ot">&lt;-</span> <span class="fu">autoThreshold</span>(x, <span class="at">est =</span> .<span class="dv">002</span>)</span>
<span id="cb2-4"><a href="#cb2-4"></a>  x[x <span class="sc">&lt;</span> th[<span class="dv">5</span>]] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb2-5"><a href="#cb2-5"></a>  <span class="fu">return</span>(x)</span>
<span id="cb2-6"><a href="#cb2-6"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="image-processing-function" class="level2" data-number="4">
<h2 data-number="4" class="anchored" data-anchor-id="image-processing-function"><span class="header-section-number">4</span> Image Processing Function</h2>
<p>This function takes a file path, an optional trim amount, an optional sample size, and an optional z-distance (the distance between slices in the cell). It then reads in the stacked image, and trims out the noisy slices (if a trim amount is supplied). It then applies a thresholding algorithm to each slice individually (see the helper function above). Next the image slices are turned into 3D-points, with a 4th dimension containing the original pixel intensity. A sample is then taken from this point cloud, and the persistence diagram is computed from this sample.</p>
<p>The function returns a list containing the original file path of the image, the raw slices, the trimmed raw slices, the thresholded slices, the point cloud, the sample point cloud, a 3D plot of the sample point cloud, and the persistence diagram.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode numberSource R code number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>processImageStack <span class="ot">&lt;-</span> <span class="cf">function</span>(fp, <span class="at">trmAmnt =</span> <span class="cn">NULL</span>, <span class="at">smpSize =</span> <span class="cn">NULL</span>, <span class="at">zDist =</span> <span class="cn">NULL</span>)</span>
<span id="cb3-2"><a href="#cb3-2"></a>{</span>
<span id="cb3-3"><a href="#cb3-3"></a>  <span class="do">## Read in the stacked image, then trim out the noise (if told to).</span></span>
<span id="cb3-4"><a href="#cb3-4"></a>  rawSlices <span class="ot">&lt;-</span> <span class="fu">readTIFF</span>(fp, <span class="at">indexed =</span> <span class="cn">TRUE</span>, <span class="at">native =</span> <span class="cn">FALSE</span>, <span class="at">all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-5"><a href="#cb3-5"></a>  <span class="cf">if</span>(<span class="fu">is.null</span>(trmAmnt)) trmAmnt <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(rawSlices)</span>
<span id="cb3-6"><a href="#cb3-6"></a>  rawSlicesTrimmed <span class="ot">&lt;-</span> rawSlices[trmAmnt]</span>
<span id="cb3-7"><a href="#cb3-7"></a></span>
<span id="cb3-8"><a href="#cb3-8"></a>  <span class="do">## Autothreshold each image slice seperatly</span></span>
<span id="cb3-9"><a href="#cb3-9"></a>  thSlices <span class="ot">&lt;-</span> <span class="fu">lapply</span>(rawSlicesTrimmed, autoTH)</span>
<span id="cb3-10"><a href="#cb3-10"></a></span>
<span id="cb3-11"><a href="#cb3-11"></a>  <span class="do">## Convert all image slices (matricies) to point sets, rename z -&gt; intensity</span></span>
<span id="cb3-12"><a href="#cb3-12"></a>  pntSlices <span class="ot">&lt;-</span> <span class="fu">lapply</span>( thSlices</span>
<span id="cb3-13"><a href="#cb3-13"></a>                     , <span class="cf">function</span>(x) <span class="fu">grid.to.xyz</span>(<span class="at">data=</span>x,<span class="at">varname =</span> <span class="st">"intensity"</span>))</span>
<span id="cb3-14"><a href="#cb3-14"></a></span>
<span id="cb3-15"><a href="#cb3-15"></a>  <span class="do">## Add in z values</span></span>
<span id="cb3-16"><a href="#cb3-16"></a>  <span class="cf">if</span>(<span class="fu">is.null</span>(zDist)) zDist <span class="ot">&lt;-</span> <span class="fl">3.87595</span> </span>
<span id="cb3-17"><a href="#cb3-17"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(pntSlices))</span>
<span id="cb3-18"><a href="#cb3-18"></a>  {</span>
<span id="cb3-19"><a href="#cb3-19"></a>    pntSlices[[i]]<span class="sc">$</span>z <span class="ot">=</span> <span class="fu">rep</span>((i<span class="dv">-1</span>)<span class="sc">*</span>zDist)</span>
<span id="cb3-20"><a href="#cb3-20"></a>  }</span>
<span id="cb3-21"><a href="#cb3-21"></a></span>
<span id="cb3-22"><a href="#cb3-22"></a>  <span class="do">## Concat the lists of points, then switch order so z is before intensity</span></span>
<span id="cb3-23"><a href="#cb3-23"></a>  pointCloud <span class="ot">&lt;-</span> <span class="fu">do.call</span>(<span class="st">"rbind"</span>, pntSlices)</span>
<span id="cb3-24"><a href="#cb3-24"></a>  pointCloud <span class="ot">&lt;-</span> pointCloud[, <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">4</span>,<span class="dv">3</span>)]</span>
<span id="cb3-25"><a href="#cb3-25"></a></span>
<span id="cb3-26"><a href="#cb3-26"></a></span>
<span id="cb3-27"><a href="#cb3-27"></a>  <span class="do">## Get a point sample</span></span>
<span id="cb3-28"><a href="#cb3-28"></a>  <span class="cf">if</span>(<span class="fu">is.null</span>(smpSize)) smpSize <span class="ot">&lt;-</span> <span class="dv">500</span></span>
<span id="cb3-29"><a href="#cb3-29"></a>  sPointCloud <span class="ot">&lt;-</span> pointCloud[ <span class="fu">sample</span>( <span class="fu">nrow</span>(pointCloud)</span>
<span id="cb3-30"><a href="#cb3-30"></a>                                   , smpSize</span>
<span id="cb3-31"><a href="#cb3-31"></a>                                   , <span class="at">prob =</span> pointCloud<span class="sc">$</span>intensity), ]</span>
<span id="cb3-32"><a href="#cb3-32"></a></span>
<span id="cb3-33"><a href="#cb3-33"></a>  <span class="do">## Generate the plot</span></span>
<span id="cb3-34"><a href="#cb3-34"></a>  x <span class="ot">&lt;-</span> sPointCloud<span class="sc">$</span>x</span>
<span id="cb3-35"><a href="#cb3-35"></a>  y <span class="ot">&lt;-</span> sPointCloud<span class="sc">$</span>y</span>
<span id="cb3-36"><a href="#cb3-36"></a>  z <span class="ot">&lt;-</span> sPointCloud<span class="sc">$</span>z</span>
<span id="cb3-37"><a href="#cb3-37"></a>  cloudPlot <span class="ot">&lt;-</span> {<span class="fu">plot_ly</span>(</span>
<span id="cb3-38"><a href="#cb3-38"></a>    <span class="at">x=</span>x, <span class="at">y=</span>y, <span class="at">z=</span>z, </span>
<span id="cb3-39"><a href="#cb3-39"></a>    <span class="at">type=</span><span class="st">"scatter3d"</span>, </span>
<span id="cb3-40"><a href="#cb3-40"></a>    <span class="at">mode=</span><span class="st">"markers"</span>, </span>
<span id="cb3-41"><a href="#cb3-41"></a>    <span class="at">text=</span><span class="fu">paste</span>(<span class="st">"Intensity: "</span>, sPointCloud<span class="sc">$</span>intensity),</span>
<span id="cb3-42"><a href="#cb3-42"></a>    <span class="at">marker=</span><span class="fu">list</span>(<span class="at">opacity=</span>.<span class="dv">6</span>, <span class="at">color=</span>sPointCloud<span class="sc">$</span>intensity)</span>
<span id="cb3-43"><a href="#cb3-43"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb3-44"><a href="#cb3-44"></a>  <span class="fu">layout</span>(<span class="at">title =</span> fp)</span>
<span id="cb3-45"><a href="#cb3-45"></a>  }</span>
<span id="cb3-46"><a href="#cb3-46"></a></span>
<span id="cb3-47"><a href="#cb3-47"></a>  <span class="do">## Compute the persistence diagram, only for the spacial coordinates</span></span>
<span id="cb3-48"><a href="#cb3-48"></a>  sRipsDiag <span class="ot">&lt;-</span> {</span>
<span id="cb3-49"><a href="#cb3-49"></a>    <span class="fu">ripsDiag</span>(</span>
<span id="cb3-50"><a href="#cb3-50"></a>      sPointCloud[ ,<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>],</span>
<span id="cb3-51"><a href="#cb3-51"></a>      <span class="at">maxdimension =</span> <span class="dv">2</span>,</span>
<span id="cb3-52"><a href="#cb3-52"></a>      <span class="at">maxscale =</span> <span class="dv">40</span>, </span>
<span id="cb3-53"><a href="#cb3-53"></a>      <span class="at">library =</span> <span class="st">"GUDHI"</span></span>
<span id="cb3-54"><a href="#cb3-54"></a>    )</span>
<span id="cb3-55"><a href="#cb3-55"></a>  }</span>
<span id="cb3-56"><a href="#cb3-56"></a></span>
<span id="cb3-57"><a href="#cb3-57"></a>  <span class="do">## Generate our return list</span></span>
<span id="cb3-58"><a href="#cb3-58"></a>  retList <span class="ot">&lt;-</span> {<span class="fu">list</span>(</span>
<span id="cb3-59"><a href="#cb3-59"></a>    <span class="st">"filePath"</span>         <span class="ot">=</span> fp,</span>
<span id="cb3-60"><a href="#cb3-60"></a>    <span class="st">"rawSlices"</span>        <span class="ot">=</span> rawSlices, </span>
<span id="cb3-61"><a href="#cb3-61"></a>    <span class="st">"rawSlicesTrimmed"</span> <span class="ot">=</span> rawSlicesTrimmed,</span>
<span id="cb3-62"><a href="#cb3-62"></a>    <span class="st">"thSlices"</span>         <span class="ot">=</span> thSlices,</span>
<span id="cb3-63"><a href="#cb3-63"></a>    <span class="st">"pointCloud"</span>       <span class="ot">=</span> pointCloud,</span>
<span id="cb3-64"><a href="#cb3-64"></a>    <span class="st">"sPointCloud"</span>      <span class="ot">=</span> sPointCloud,</span>
<span id="cb3-65"><a href="#cb3-65"></a>    <span class="st">"cloudPlot"</span>        <span class="ot">=</span> cloudPlot,</span>
<span id="cb3-66"><a href="#cb3-66"></a>    <span class="st">"ripsDiag"</span>         <span class="ot">=</span> sRipsDiag</span>
<span id="cb3-67"><a href="#cb3-67"></a>  )}</span>
<span id="cb3-68"><a href="#cb3-68"></a>  <span class="fu">return</span>(retList)</span>
<span id="cb3-69"><a href="#cb3-69"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="test-image" class="level2" data-number="5">
<h2 data-number="5" class="anchored" data-anchor-id="test-image"><span class="header-section-number">5</span> Test Image</h2>
<p>Now lets run this on a test image.</p>
<section id="read-and-process-the-image" class="level3" data-number="5.1">
<h3 data-number="5.1" class="anchored" data-anchor-id="read-and-process-the-image"><span class="header-section-number">5.1</span> Read and Process the Image</h3>
<div class="sourceCode" id="cb4"><pre class="sourceCode numberSource R code number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>testFP  <span class="ot">&lt;-</span> <span class="st">"test.tif"</span></span>
<span id="cb4-2"><a href="#cb4-2"></a>testT   <span class="ot">&lt;-</span> <span class="dv">14</span><span class="sc">:</span><span class="dv">36</span></span>
<span id="cb4-3"><a href="#cb4-3"></a>testR   <span class="ot">&lt;-</span> <span class="fu">processImageStack</span>(<span class="at">fp =</span> testFP, <span class="at">trmAmnt =</span> testT, <span class="at">smpSize =</span> <span class="dv">600</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="results" class="level3" data-number="5.2">
<h3 data-number="5.2" class="anchored" data-anchor-id="results"><span class="header-section-number">5.2</span> Results</h3>
<section id="raw-image-slices" class="level4" data-number="5.2.1">
<h4 data-number="5.2.1" class="anchored" data-anchor-id="raw-image-slices"><span class="header-section-number">5.2.1</span> Raw Image Slices</h4>
<p>This is a heat map of each slice of the cell in its original state, no filtering, thresholding, or trimming has been applied at this point.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode numberSource R code number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="fu">image2D</span>(testR<span class="sc">$</span>rawSlices,<span class="at">asp =</span> <span class="dv">1</span>, <span class="at">axes=</span><span class="cn">FALSE</span>, <span class="at">colkey =</span> <span class="cn">FALSE</span>, <span class="at">xlab=</span><span class="st">""</span>, <span class="at">ylab=</span><span class="st">""</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src=".ob-jupyter/4e37ec94b1be55a695349f73b8f01f88841854fd.png" class="img-fluid"> <img src=".ob-jupyter/0fb00e44b437ed37b8d474744e0aa4898c57fdb8.png" class="img-fluid"> <img src=".ob-jupyter/9de30cd19ffc2055a145f1a78d597b1f82c44d79.png" class="img-fluid"> <img src=".ob-jupyter/5fcf38770633aaf2f12a24225b3fc5ade2cd292e.png" class="img-fluid"> <img src=".ob-jupyter/82fe62a88a25059648f541fa6c79a371ab008b69.png" class="img-fluid"></p>
</section>
<section id="raw-trimmed-image-slices" class="level4" data-number="5.2.2">
<h4 data-number="5.2.2" class="anchored" data-anchor-id="raw-trimmed-image-slices"><span class="header-section-number">5.2.2</span> Raw Trimmed Image Slices</h4>
<p>Since the first few and last few slices appear to be nothing but noise, they have been trimmed out, and the results are shown below. The results from computing the homology are much better when this is done. At this point the trimming has been done manually based on each image, but it would be nice to have an automated way to do it.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode numberSource R code number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="fu">image2D</span>(testR<span class="sc">$</span>rawSlicesTrimmed,<span class="at">asp =</span> <span class="dv">1</span>, <span class="at">axes=</span><span class="cn">FALSE</span>, <span class="at">colkey =</span> <span class="cn">FALSE</span>, <span class="at">xlab=</span><span class="st">""</span>, <span class="at">ylab=</span><span class="st">""</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src=".ob-jupyter/3b73763340e672c5332d2e3637b59182b63bad8a.png" class="img-fluid"> <img src=".ob-jupyter/dd530524401550024d7b00227d4fbfa52f908173.png" class="img-fluid"> <img src=".ob-jupyter/40c58d231513855eeb14cc1d98327793182fbb14.png" class="img-fluid"></p>
</section>
<section id="thresholded-image-slices" class="level4" data-number="5.2.3">
<h4 data-number="5.2.3" class="anchored" data-anchor-id="thresholded-image-slices"><span class="header-section-number">5.2.3</span> Thresholded Image Slices</h4>
<p>Next we show the results after an automatic thresholding function has been applied to each slice individually. This function comes from an R-package.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode numberSource R code number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="fu">image2D</span>(testR<span class="sc">$</span>thSlices,<span class="at">asp =</span> <span class="dv">1</span>, <span class="at">axes=</span><span class="cn">FALSE</span>, <span class="at">colkey =</span> <span class="cn">FALSE</span>, <span class="at">xlab=</span><span class="st">""</span>, <span class="at">ylab=</span><span class="st">""</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src=".ob-jupyter/313d2b0b229d460552aaeb46bc04de9d9ad13390.png" class="img-fluid"> <img src=".ob-jupyter/466e15f46d42003a6f6c549834c847ca40aac60d.png" class="img-fluid"> <img src=".ob-jupyter/80da1612f3a1e057ad402f79dbbb87f841d043aa.png" class="img-fluid"></p>
</section>
<section id="d-point-cloud" class="level4" data-number="5.2.4">
<h4 data-number="5.2.4" class="anchored" data-anchor-id="d-point-cloud"><span class="header-section-number">5.2.4</span> 3D Point Cloud</h4>
<p>Below is an interact 3D point cloud of a random sample taken from the full point cloud generated from the stacked image. The sample is random, but points with higher intensity have a higher probability of being chosen. Hence if there is a high intensity region in the cell, there is a high probability that the sample will include points from this area.</p>
<div class="sourceCode" id="cb8" data-file="test-3d-point-cloud.html" data-exports="code"><pre class="sourceCode numberSource R code number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>testR<span class="sc">$</span>cloudPlot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><a href="test-3d-point-cloud.html">file:test-3d-point-cloud.html</a></p>
</section>
<section id="the-persistence-diagram" class="level4" data-number="5.2.5">
<h4 data-number="5.2.5" class="anchored" data-anchor-id="the-persistence-diagram"><span class="header-section-number">5.2.5</span> The Persistence Diagram</h4>
<p>Here is the persistence diagram for the test image. The black dots are the connected components, the red are the 1-holes, and the blue are the 2-holes. For a cell with high intensity around the boundary, one would expect a blue point far away from the diagonal. For a cell with intensity dispersed throughout, one would expect any blue points to be near the diagonal.</p>
<div class="sourceCode" id="cb9" data-file="test-persistence-diagram.png"><pre class="sourceCode numberSource R code number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb9-2"><a href="#cb9-2"></a><span class="fu">plot</span>(testR<span class="sc">$</span>ripsDiag[[<span class="st">"diagram"</span>]])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="test-persistence-diagram.png" class="img-fluid"></p>
</section>
</section>
</section>
<section id="high-intensity-on-the-cell-boundary" class="level2" data-number="6">
<h2 data-number="6" class="anchored" data-anchor-id="high-intensity-on-the-cell-boundary"><span class="header-section-number">6</span> High Intensity on the Cell Boundary</h2>
<p>Here will will look at a cell with high intensity around the cell boundary. For this example, one would expect the persistence diagram to have one blue point far away from the diagonal.</p>
<section id="read-and-process-the-image-1" class="level3" data-number="6.1">
<h3 data-number="6.1" class="anchored" data-anchor-id="read-and-process-the-image-1"><span class="header-section-number">6.1</span> Read and Process the Image</h3>
<div class="sourceCode" id="cb10"><pre class="sourceCode numberSource R code number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>strongFP <span class="ot">&lt;-</span> <span class="st">"Boundary-Strong.tif"</span></span>
<span id="cb10-2"><a href="#cb10-2"></a>strongT  <span class="ot">&lt;-</span> <span class="dv">15</span><span class="sc">:</span><span class="dv">38</span></span>
<span id="cb10-3"><a href="#cb10-3"></a>strongR  <span class="ot">&lt;-</span> <span class="fu">processImageStack</span>(<span class="at">fp =</span> strongFP, <span class="at">trmAmnt =</span> strongT)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="raw-image-slices-1" class="level3" data-number="6.2">
<h3 data-number="6.2" class="anchored" data-anchor-id="raw-image-slices-1"><span class="header-section-number">6.2</span> Raw Image Slices</h3>
<div class="sourceCode" id="cb11"><pre class="sourceCode numberSource R code number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a><span class="fu">image2D</span>(strongR<span class="sc">$</span>rawSlices,<span class="at">asp =</span> <span class="dv">1</span>, <span class="at">axes=</span><span class="cn">FALSE</span>, <span class="at">colkey =</span> <span class="cn">FALSE</span>, <span class="at">xlab=</span><span class="st">""</span>, <span class="at">ylab=</span><span class="st">""</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src=".ob-jupyter/4e37ec94b1be55a695349f73b8f01f88841854fd.png" class="img-fluid"> <img src=".ob-jupyter/0fb00e44b437ed37b8d474744e0aa4898c57fdb8.png" class="img-fluid"> <img src=".ob-jupyter/9de30cd19ffc2055a145f1a78d597b1f82c44d79.png" class="img-fluid"> <img src=".ob-jupyter/5fcf38770633aaf2f12a24225b3fc5ade2cd292e.png" class="img-fluid"> <img src=".ob-jupyter/82fe62a88a25059648f541fa6c79a371ab008b69.png" class="img-fluid"></p>
</section>
<section id="d-point-cloud-1" class="level3" data-number="6.3">
<h3 data-number="6.3" class="anchored" data-anchor-id="d-point-cloud-1"><span class="header-section-number">6.3</span> 3D Point Cloud</h3>
<div class="sourceCode" id="cb12" data-file="high-intensity-boundary.html" data-exports="code"><pre class="sourceCode numberSource R code number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>strongR<span class="sc">$</span>cloudPlot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><a href="high-intensity-boundary.html">file:high-intensity-boundary.html</a></p>
</section>
<section id="the-persistence-diagram-1" class="level3" data-number="6.4">
<h3 data-number="6.4" class="anchored" data-anchor-id="the-persistence-diagram-1"><span class="header-section-number">6.4</span> The Persistence Diagram</h3>
<div class="sourceCode" id="cb13" data-file="high-intensity-boundary-persistence-diagram.png"><pre class="sourceCode numberSource R code number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb13-2"><a href="#cb13-2"></a><span class="fu">plot</span>(strongR<span class="sc">$</span>ripsDiag[[<span class="st">"diagram"</span>]])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="high-intensity-boundary-persistence-diagram.png" class="img-fluid"></p>
</section>
</section>
<section id="medium-intensity-on-the-cell-boundary" class="level2" data-number="7">
<h2 data-number="7" class="anchored" data-anchor-id="medium-intensity-on-the-cell-boundary"><span class="header-section-number">7</span> Medium Intensity on the Cell Boundary</h2>
<p>Here will will look at a cell with medium intensity around the cell boundary. For this example, one would expect the persistence diagram to have one blue point a moderate distance away from the diagonal.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode numberSource R code number-lines code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a>mediumFP <span class="ot">&lt;-</span> <span class="st">"Boundary-Medium.tif"</span></span>
<span id="cb14-2"><a href="#cb14-2"></a>mediumT  <span class="ot">&lt;-</span> <span class="dv">7</span><span class="sc">:</span><span class="dv">29</span></span>
<span id="cb14-3"><a href="#cb14-3"></a>mediumR  <span class="ot">&lt;-</span> <span class="fu">processImageStack</span>(<span class="at">fp =</span> mediumFP, <span class="at">trmAmnt =</span> mediumT)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb15"><pre class="sourceCode numberSource R code number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a><span class="fu">image2D</span>(mediumR<span class="sc">$</span>rawSlices,<span class="at">asp =</span> <span class="dv">1</span>, <span class="at">axes=</span><span class="cn">FALSE</span>, <span class="at">colkey =</span> <span class="cn">FALSE</span>, <span class="at">xlab=</span><span class="st">""</span>, <span class="at">ylab=</span><span class="st">""</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src=".ob-jupyter/6a72aeb4cfa922b1a13d8ce307c4c35f4ff9e1fe.png" class="img-fluid"> <img src=".ob-jupyter/23d6e5932d96225e8e90b62ee260c9f81f367bdd.png" class="img-fluid"> <img src=".ob-jupyter/9f64bfb84a3da40403ae2aa24bb56114b7bbb36c.png" class="img-fluid"> <img src=".ob-jupyter/0f9aa8506b7a7e987089b7ce8a5b7e111d9ef6b9.png" class="img-fluid"> <img src=".ob-jupyter/c68966819c74c8a9e898b1afab649b542d8d7e6e.png" class="img-fluid"></p>
<div class="sourceCode" id="cb16" data-file="medium-intensity-boundary.html" data-exports="code"><pre class="sourceCode numberSource R code number-lines code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a>mediumR<span class="sc">$</span>cloudPlot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><a href="medium-intensity-boundary.html">file:medium-intensity-boundary.html</a></p>
<div class="sourceCode" id="cb17" data-file="medium-intensity-boundary-persistence-diagram.png"><pre class="sourceCode numberSource R code number-lines code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb17-2"><a href="#cb17-2"></a><span class="fu">plot</span>(mediumR<span class="sc">$</span>ripsDiag[[<span class="st">"diagram"</span>]])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="medium-intensity-boundary-persistence-diagram.png" class="img-fluid"></p>
</section>
<section id="low-intensity-on-the-cell-boundary" class="level2" data-number="8">
<h2 data-number="8" class="anchored" data-anchor-id="low-intensity-on-the-cell-boundary"><span class="header-section-number">8</span> Low Intensity on the Cell Boundary</h2>
<p>Here will will look at a cell with low intensity around the cell boundary. For this example, one would expect the persistence diagram to have no blue points except near the diagonal.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode numberSource R code number-lines code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a>weakFP <span class="ot">&lt;-</span> <span class="st">"Boundary-Weak.tif"</span></span>
<span id="cb18-2"><a href="#cb18-2"></a>weakT  <span class="ot">&lt;-</span> <span class="dv">13</span><span class="sc">:</span><span class="dv">29</span></span>
<span id="cb18-3"><a href="#cb18-3"></a>weakR  <span class="ot">&lt;-</span> <span class="fu">processImageStack</span>(<span class="at">fp =</span> weakFP, <span class="at">trmAmnt =</span> weakT)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb19"><pre class="sourceCode numberSource R code number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a><span class="fu">image2D</span>(weakR<span class="sc">$</span>rawSlices,<span class="at">asp =</span> <span class="dv">1</span>, <span class="at">axes=</span><span class="cn">FALSE</span>, <span class="at">colkey =</span> <span class="cn">FALSE</span>, <span class="at">xlab=</span><span class="st">""</span>, <span class="at">ylab=</span><span class="st">""</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src=".ob-jupyter/72575b13c488ace9b29a98865952954bcc756d08.png" class="img-fluid"> <img src=".ob-jupyter/3a568d6cf8e7ec4fb461f3d60eabf45aa466b73c.png" class="img-fluid"> <img src=".ob-jupyter/83d94720740ebd19bd84c6ae8251ebf22c4ac13c.png" class="img-fluid"> <img src=".ob-jupyter/48393dc450d10457a99295fae2ae25c9de7bdbef.png" class="img-fluid"> <img src=".ob-jupyter/83d4ebef82cb3c1dec43e774c3815824f072b087.png" class="img-fluid"></p>
<div class="sourceCode" id="cb20" data-file="weak-intensity-boundary.html" data-exports="code"><pre class="sourceCode numberSource R code number-lines code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a>weakR<span class="sc">$</span>cloudPlot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><a href="weak-intensity-boundary.html">file:weak-intensity-boundary.html</a></p>
<div class="sourceCode" id="cb21" data-file="weak-intensity-boundary.png"><pre class="sourceCode numberSource R code number-lines code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb21-2"><a href="#cb21-2"></a><span class="fu">plot</span>(weakR<span class="sc">$</span>ripsDiag[[<span class="st">"diagram"</span>]])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="weak-intensity-boundary.png" class="img-fluid"></p>
</section>
<section id="results-comparison" class="level2" data-number="9">
<h2 data-number="9" class="anchored" data-anchor-id="results-comparison"><span class="header-section-number">9</span> Results Comparison</h2>
<div class="sourceCode" id="cb22" data-file="comparison.png"><pre class="sourceCode numberSource R code number-lines code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">1</span>))</span>
<span id="cb22-2"><a href="#cb22-2"></a><span class="fu">plot</span>(weakR<span class="sc">$</span>ripsDiag[[<span class="st">"diagram"</span>]], <span class="at">main=</span><span class="st">"Weak Intensity Boundary"</span>)</span>
<span id="cb22-3"><a href="#cb22-3"></a><span class="fu">plot</span>(mediumR<span class="sc">$</span>ripsDiag[[<span class="st">"diagram"</span>]], <span class="at">main=</span><span class="st">"Medium Intensity Boundary"</span>)</span>
<span id="cb22-4"><a href="#cb22-4"></a><span class="fu">plot</span>(strongR<span class="sc">$</span>ripsDiag[[<span class="st">"diagram"</span>]], <span class="at">main=</span><span class="st">"Strong Intensity Boundary"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="RESULTS drawer">
<p><img src="comparison.png" class="img-fluid"></p>
</div>
<p>The weak intensity boundary is indicative of the medication working effectively. The blue dots all begin close to the diagonal in the persistence diagram indicates that the protein distribution has no voids and is evenly distributed in the cell.</p>
<p>The strong intensity boundary is indicative of the medication not working effectively. The blue dot far from the diagonal in the persistence diagrams indicates that the protein is distributed along the cell boundary, meaning little protein made it into the cell.</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var filterRegex = new RegExp("https:\/\/test-mathnotes\.netlify\.app");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
          // target, if specified
          link.setAttribute("target", "_blank");
          // default icon
          link.classList.add("external");
      }
    }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../../projects/index.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Research Projects</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../../projects/datasaurus/index.html" class="pagination-link">
        <span class="nav-page-text">Datasaurus</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">This page is built with <a href="https://quarto.org">Quarto</a>.</div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right"> 2023 Ryan Jensen</div>
  </div>
</footer>



</body></html>